metadata-ingestion/src/datahub/ingestion/source/qlik_cloud/qlik_api.py

```diff
@@ -0,0 +1,144 @@
+import logging
+import sys
+from datetime import datetime
+from typing import Any, List, Optional
+
+import requests
+
+from datahub.ingestion.source.qlik_cloud.config import (
+    QLIK_DATETIME_FORMAT,
+    Constant,
+    QlikSourceConfig,
+)
+from datahub.ingestion.source.qlik_cloud.data_classes import (
+    App,
+    Item,
+    QlikDataset,
+    SchemaField,
+    Space,
+)
+
+# Logger instance
+logger = logging.getLogger(__name__)
+
+
+class QlikAPI:
+    def __init__(self, config: QlikSourceConfig) -> None:
+        self.config = config
+        self.session = requests.Session()
+        self.session.headers.update(
+            {
+                "Authorization": f"Bearer {self.config.api_key}",
+                "Content-Type": "application/json",
+            }
+        )
+        self.base_url = f"{self.config.tenant_hostname}/api/v1"
+        # Test connection by fetching list of api keys
+        logger.info("Trying to connect to {}".format(self.base_url))
+        self.session.get(f"{self.base_url}/api-keys").raise_for_status()
+
+    def log_http_error(self, message: str) -> Any:
+        logger.warning(message)
+        _, e, _ = sys.exc_info()
+        if isinstance(e, requests.exceptions.HTTPError):
+            logger.warning(f"HTTP status-code = {e.response.status_code}")
+        logger.debug(msg=message, exc_info=e)
+        return e
+
+    def get_spaces(self) -> List[Space]:
+        spaces: List[Space] = []
+        try:
+            response = self.session.get(f"{self.base_url}/spaces")
+            response.raise_for_status()
+            for space in response.json()[Constant.DATA]:
+                spaces.append(
+                    Space(
```

Comment:
let's change Space (and other data class types) to be pydantic models (e.g. inherit from pydantic.BaseModel)

then we can use `Space.parse_obj(...)`, and avoid needing all these constants